nincat

```diff
@@ -73,7 +73,7 @@ print_file () {
   # Bash has some specialities that makes this program's life harder.
   # Famous bashishms it adds things that shouldn't for the shell and to get it out of the way,
   # we have to change it works under the hood.
-  [ "$(ps -hp $$ | awk '{print $5}')" = "bash" ] && shopt -s xpg_echo
+  grep -q "$(ps -hp $$ | awk '{print $5}')" <<<"bash" && shopt -s xpg_echo
```

Comment:
`<<<` is built-in bashism, it won't work on some shells such as `dash`.

A pipe-call with grep should handle better the job:

```bash
[ "$(ps -hp $$ | awk '$5 ~ /bash/ {print 1}')" ] && shopt -s xpg_echo
```

Give it a try on your system and let's see if works.

Reply:
Oh sorry i didn't know about that. This method doesn't seem to work on my setup as `$$` consistently returns `nincat` . I found [this simple method on StackExchange](https://unix.stackexchange.com/a/72475) , i don't think it would have much side effect on other shells : `[ -n "$BASH_VERSION" ] && shopt -s xpg_echo` . What do you think ?

Reply:
It works indeed, can you commit it so I can approve this PR?

Reply:
done ! works perfectly on my end